pmlc/ast/builder.cc

```diff
@@ -738,6 +739,38 @@ struct ProgramBuilder {
         .result();
   }
 
+  Value makeGatherOp(ExprNodeIntrinsic *node, ArrayRef<Value> operands) {
+    TensorShape shape = evaluator.getShape(node);
+    RankedTensorType resultType = builder.getRankedTensorType(shape);
+    auto op = builder.create<tile::GatherOp>(
+        loc, resultType, ValueRange{operands[0], operands[1]},
+        ArrayRef<NamedAttribute>{});
+
+    FloatAttr cubicCoeff;
+    if (!m_Constant(&cubicCoeff).match(operands[4].getDefiningOp())) {
+      throw std::runtime_error(
+          "'cubicCoeff' primitive expects argument 5 to be a constant float");
+    }
+    auto cubicCoeffFloat = static_cast<float>(cubicCoeff.getValueAsDouble());
+    op.setAttr("cubicCoeff", builder.getF32FloatAttr(cubicCoeffFloat));
+
+    IntegerAttr mode;
+    if (!m_Constant(&mode).match(operands[3].getDefiningOp())) {
+      throw std::runtime_error(
+          "'mode' primitive expects argument 4 to be a constant integer");
+    }
+    op.setAttr("mode", mode);
+
+    IntegerAttr axis;
+    if (!m_Constant(&axis).match(operands[2].getDefiningOp())) {
+      throw std::runtime_error(
+          "'axis' primitive expects argument 3 to be a constant integer");
+    }
+    op.setAttr("axis", builder.getIndexAttr(axis.getInt()));
```

Comment:
I think it'd be better if we had a single builder function that took all the parameters during construction of the op instead of setting the attributes after constructing the op.

Reply:
Replaced with a single builder function.